pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -89,6 +89,108 @@ func getApplicablePodNameForAppFramework(cr splcommon.MetaObject, ordinalIdx int
 	return fmt.Sprintf("splunk-%s-%s-%d", cr.GetName(), podType, ordinalIdx)
 }
 
+// runCustomCommandOnSplunkPods  runs the specified custom command on the pod/s
+func runCustomCommandOnSplunkPods(ctx context.Context, cr splcommon.MetaObject, replicas int32, command string, podExecClient splutil.PodExecClientImpl) error {
+	var err error
+	var stdOut string
+
+	streamOptions := splutil.NewStreamOptionsObject(command)
+	// Run the command on each replica pod
+	for replicaIndex := 0; replicaIndex < int(replicas); replicaIndex++ {
+		// get the target pod name
+		podName := getApplicablePodNameForAppFramework(cr, replicaIndex)
+		podExecClient.SetTargetPodName(ctx, podName)
+
+		// CSPL-1639: reset the Stdin so that reader pipe can read from the correct offset of the string reader.
+		// This is particularly needed in the cases where we are trying to run the same command across multiple pods
+		// and we need to clear the reader pipe so that we can read the read buffer from the correct offset again.
+		splutil.ResetStringReader(streamOptions, command)
+
+		// Throw an error if we are not able to run the command
+		stdOut, _, err = podExecClient.RunPodExecCommand(ctx, streamOptions, []string{"/bin/sh"})
+		if err != nil {
+			err = fmt.Errorf("unable to run command %s. stdout: %s, err: %s", command, stdOut, err)
+			break
+		}
+	}
+	return err
+}
+
+// Get extension for name of telemetry app
+func getTelAppNameExtension(crKind string) (string, error) {
+	switch crKind {
+	case "Standalone":
+		return "stdaln", nil
+	case "LicenseMaster":
+		return "lm", nil
+	case "SearchHeadCluster":
+		return "shc", nil
+	case "ClusterMaster":
+		return "cm", nil
+	default:
+		return "", errors.New("Invalid CR kind for telemetry app")
+	}
+}
+
+// addTelApp adds a telemetry app
+var addTelApp = func(ctx context.Context, podExecClient splutil.PodExecClientImpl, replicas int32, cr splcommon.MetaObject) error {
```

Comment:
We are using function pointers to opt out of executing the function for UT instead of using the 'mock' flag.

Reply:
Well that is what mockPodExecClient was for in the first place, so that you can still execute the function with mock client and still cover the lines in code coverage.

Reply:
I have used mockPodExecClient for testing the function **addTelApp**. But when the call comes from UT's calling say **ApplyClusterManager**, **ApplyLicenseManager** wherein the status of the CR reaches a **Ready** state, we are using this method to avoid calling **addTelApp** as we cannot pass **mockPodExecClient** as a client there.

Reply:
Ok got it 👍